src/util/operations.spec.tsx

```diff
@@ -44,3 +44,30 @@ describe("getInstanceName", () => {
     expect(name).toBe("testInstance3");
   });
 });
+
+describe("getProjectName", () => {
+  it("identifies project name from an instance operation when no project parameter is present", () => {
+    const operation = craftOperation("/1.0/instances/testInstance1");
+    const name = getProjectName(operation);
+
+    expect(name).toBe("default");
+  });
+
+  it("identifies project name from an instance operation in a custom project", () => {
+    const operation = craftOperation(
+      "/1.0/instances/testInstance2?project=fooProject"
+    );
+    const name = getProjectName(operation);
+
+    expect(name).toBe("fooProject");
+  });
+
+  it("identifies project name from an instance operation in a custom project with other parameters", () => {
+    const operation = craftOperation(
+      "/1.0/instances/testInstance2?foo=bar&project=barProject"
+    );
+    const name = getProjectName(operation);
+
+    expect(name).toBe("barProject");
+  });
+});
```

Comment:
Thanks for extending the cases and fixing the implementation. I think there is still an uncovered edge case with this url: `/1.0/instances/testInstance1?foo=bar&project=barProject&baz=qux`

Reply:
Fixed it. Thanks :)